src/agent/src/mount.rs

```diff
@@ -486,17 +486,8 @@ fn mount_storage(logger: &Logger, storage: &Storage) -> Result<()> {
         return Ok(());
     }
 
-    match storage.fstype.as_str() {
-        DRIVER_9P_TYPE | DRIVER_VIRTIOFS_TYPE => {
-            let dest_path = Path::new(storage.mount_point.as_str());
-            if !dest_path.exists() {
-                fs::create_dir_all(dest_path).context("Create mount destination failed")?;
-            }
-        }
-        _ => {
-            ensure_destination_exists(storage.mount_point.as_str(), storage.fstype.as_str())?;
-        }
-    }
+    let mount_path = Path::new(&storage.mount_point);
+    fs::create_dir_all(mount_path).context("Could not create mountpoint")?;
```

Comment:
Actually we did have the file based volume such as  -v /dev/sdX:<container dir>, for this case, kata would hot plug this block device into guest, and agent wouldn't mount it with its real filesystem, but instead it only did the "bind" mount of this device in guest and in container.

So I think we'd better keep the ensure_destination_exists and enhance it to deal with the file  based storage case correctly.

Reply:
What exactly does -v /dev/sdX:<dir> do>?  Does that mount the block device at /dev/sdX as a filesystem in the container?

If so it's not actually a file volume in the sense we mean here: we still need a directory mountpoint.  The only case where we need a file mountpoint is where it's a bind mount *and* the mount target is a file rather than a directory.

Reply:
Hi @dgibson  For -v /dev/sdX:<dir>, kata would hotplug the corresponding device into guest, but  since runtime couldn’t know what’s block device’s real filesystem, thus it only do the bind mount of the device file  to an temp destination in guest instead of do the real filesystem mount, thus the storage mount only did the bind mount of device file.

Reply:
I've addressed this case in my latest push